packages/vite/src/node/utils.ts

```diff
@@ -534,6 +534,38 @@ export function removeDirSync(dir: string) {
   }
 }
 
+const REMOVE_DIR_TIMEOUT = 5000
+const rmdir = promisify(fs.rm ?? fs.rmdir) // TODO: Remove after support for Node 12 is dropped
+export async function removeDir(dir: string) {
+  const t0 = Date.now()
+  let backoff = 0
+  const attempt = async () => {
+    const attemptStart = Date.now()
+    try {
+      await rmdir(dir, { recursive: true })
+    } catch (e) {
+      if (e.code === 'ENOENT') {
+        return
+      }
+      if (e.code === 'ENOTEMPTY' || e.code === 'EPERM') {
+        const now = Date.now()
+        if (now - t0 < REMOVE_DIR_TIMEOUT) {
+          const delay = backoff - (now - attemptStart)
+          if (delay > 0) {
+            await new Promise((resolve) => setTimeout(resolve, delay))
+          }
+          backoff = Math.min(backoff + 10, 100)
+          await attempt()
+          return
+        }
+      }
```

Comment:
I wonder if we should have this look similar to `gracefulRename` so we kill two birds with one stone when reading/understanding the code. It involves some calculations which takes time to consume.

Reply:
Yes, I was planning to write exactly the same and got to a meeting 😅. 
The code could even be abstracted but I think it may be better to duplicate it. But they should be exactly the same, and the wiring should also be equal (including the windows check you mentioned before)

Reply:
I've pushed the suggested change.